File: java/app/src/main/java/tech/cherri/tokenpushexample/fragment/LoginFragment.java
package tech.cherri.tokenpushexample.fragment;

import android.app.Fragment;
import android.app.FragmentManager;
import android.app.FragmentTransaction;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import tech.cherri.tokenpushexample.R;
import tech.cherri.tokenpushexample.TokenPushParam;

public class LoginFragment extends Fragment {

    private Button loginBtn;
    private TokenPushParam tokenPushParam;

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater,
                             @Nullable ViewGroup container,
                             @Nullable Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_login, container, false);
    }

    @Nullable
    @Override
    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        loginBtn = view.findViewById(R.id.login_button);
        loginBtn.setOnClickListener(v -> {
            if (tokenPushParam == null) {
                AddCardUserCancelFragment addCardUserCancelFragment = new AddCardUserCancelFragment();
                addCardUserCancelFragment.setTokenPushParam(tokenPushParam);
                FragmentManager fragmentManager = getFragmentManager();
                FragmentTransaction fragmentTransaction = fragmentManager.beginTransaction();
                fragmentTransaction.remove(LoginFragment.this);
                fragmentTransaction.replace(R.id.fragment_container_view, addCardUserCancelFragment, AddCardUserCancelFragment.class.getSimpleName());
                fragmentTransaction.commit();
            } else {
                TermsFragment termsFragment = new TermsFragment();
                termsFragment.setTokenPushParam(tokenPushParam);
                FragmentManager fragmentManager = getFragmentManager();
                FragmentTransaction fragmentTransaction = fragmentManager.beginTransaction();
                fragmentTransaction.remove(LoginFragment.this);
                fragmentTransaction.replace(R.id.fragment_container_view, termsFragment, TermsFragment.class.getSimpleName());
                fragmentTransaction.commit();
            }
        });
    }

    public TokenPushParam getTokenPushParam() {
        return tokenPushParam;
    }

    public void setTokenPushParam(TokenPushParam tokenPushParam) {
        this.tokenPushParam = tokenPushParam;
    }
}

File: kotlin/app/src/main/java/tech/cherri/tokenpushexample/fragment/LoginFragment.kt
package tech.cherri.tokenpushexample.fragment

import AddCardUserCancelFragment
import android.app.Fragment
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import tech.cherri.tokenpushexample.R
import tech.cherri.tokenpushexample.TokenPushParam

class LoginFragment : Fragment() {
    private var tokenPushParam: TokenPushParam? = null
    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        return inflater.inflate(R.layout.fragment_login, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        var loginBtn: Button = view.findViewById(R.id.login_button)
        loginBtn.setOnClickListener(View.OnClickListener { v: View? ->
            if (tokenPushParam == null) {
                val addCardUserCancelFragment = AddCardUserCancelFragment()
                addCardUserCancelFragment.setTokenPushParam(tokenPushParam)
                val fragmentManager = fragmentManager
                val fragmentTransaction = fragmentManager.beginTransaction()
                fragmentTransaction.remove(this@LoginFragment)
                fragmentTransaction.replace(
                    R.id.fragment_container_view,
                    addCardUserCancelFragment,
                    AddCardUserCancelFragment::class.java.getSimpleName()
                )
                fragmentTransaction.commit()
            } else {
                val termsFragment = TermsFragment()
                termsFragment.setTokenPushParam(tokenPushParam)
                val fragmentManager = fragmentManager
                val fragmentTransaction = fragmentManager.beginTransaction()
                fragmentTransaction.remove(this@LoginFragment)
                fragmentTransaction.replace(
                    R.id.fragment_container_view,
                    termsFragment,
                    TermsFragment::class.java.simpleName
                )
                fragmentTransaction.commit()
            }
        })
    }

    fun getTokenPushParam(): TokenPushParam? {
        return tokenPushParam
    }

    fun setTokenPushParam(tokenPushParam: TokenPushParam?) {
        this.tokenPushParam = tokenPushParam
    }
}